### DISCUSSION TITLE
How to get rewards?

### DISCUSSION THREAD

**Discussion by Black1Matter** [cosmos/community#56] [Clients (JS)] [answered]
Hello friends, I'm working on a project where I need to track blocks and look for delegations with a specific memo, then record the amount they delegated. Now, I'm facing an issue with calculating rewards. I need to receive rewards not from the entire amount delegated to a validator but only from the amount delegated with a specific memo.

I found one approach where I take the APR and use it to calculate how much the validator will receive per hour. For example, taking Cosmos APR:

Reward per hour = all_delegations_with_memo * (19.23 / 100) / 365 * 24 * 60

But it's not very precise. Perhaps you can provide some ideas on how to implement this more accurately?

**Comment by tac0turtle** [comment, marked as answer]
by watching a single account and tracking its rewards in between blocks you should be able to see the yield generated. Let me know if this works for you
